vocabulary-draft/continuous-integration-pipeline-events.md

```diff
@@ -0,0 +1,29 @@
+# Continuous Integration Pipeline Events
+
+These events are related to continuous integration pipelines, this pipelines usually include building, testing, packaging and releasing software artifacts. 
+Due the dynamic nature of Pipelines, most of actual work needs to be queued to happen in a distributed way, hence Queued events are added. 
+Adopters can choose to ignore these events if they don't apply to their use cases. 
+
+A pipeline, in the context of Continuous Integration, is the definition of a set of tasks that needs to be performed to build, test, package and release software artifacts. A pipeline can be instanciated multiple times, for example to build different versions of the same artifact. We are refering to this instance as PipelineRun. It will have a unique Id and it will help us to track the build and release progress on a particular software artifact. 
+
+- **PipelineRun Queued**: a PipelineRun has been schedule to run
+- **PipelineRun Started**: a PipelineRun has started and it is running
+- **PipelineRun Finished**: a PipelineRun has finished it execution, the event will contain the finished status, success, error or failure
+- **Build Queued**: a Build task has been queued, this build process usually is in charge of producing a binary from source code
+- **Build Started**: a Build task has started 
+- **Build Finished**: a Build task has finished, the event will contain the finished status, success, error or failure
+- **Tests Queued**: a Test task has been queued, and it is waiting to be started
+- **Tests Started**: a Test task has started
+- **Tests Finished**: a Test task has finished, the event will contain the finished status, success, error or failure
+- **Artifact Packaged**: an artifact has been packaged for distribution, this artifact is now versioned with a fixed version
+- **Artifact Released**: an artifact has been released and it can be advertised for others to use
```

Comment:
Do the names indicate any trust? When an artifact is released is this the final version that the consumer will take or do we need to capture steps in between? Example of a flow:

* The build step builds an artifact and uploads it to a central location 
* A quick test task takes the artifact and uses it to test basic things
* A long-running test task takes the artifact and uses it for long tests
* A security task takes the artifact and checks for security issues.

Do we want to capture the trust or confidence in the artifact? For example, an early adopter might want to take the artifact before going through a security scan using it in a test environment to test for stability but for the production environment, they want to wait until the security scan completes.

Reply:
@m-linner-ericsson yes.. I think that we should add that as optional metadata, for now I am taking these attributes from the events as the minimum set of metadata to represent what has happened. Inside the body of the event we can definitely include as much information as we want. But this is just my personal view. I wonder how you would capture those stages and if then we need to emit the `Artifact.Released` event multiple times with different metadata to indicate the confidence level of the artifact.

Reply:
In Eiffel, we have a concept of [Confidence Level Modified](https://github.com/eiffel-community/eiffel/blob/master/eiffel-vocabulary/EiffelConfidenceLevelModifiedEvent.md). After an artifact is build we can then send multiple confidence events for the same artifact. Thus in my example, we send 3 confidence events that link back to the built artifact. (The idea of linking events could be a discussion on its own)

Does the idea of sending other events to capture these changes sound ok?

Reply:
@m-linner-ericsson for me, (personal opinion again) this is a great thing to have, but I would delay this discussion for later after we have the first set of events defined at a high-level. The approach is to define the minimum list of events that allows us to explain the benefit of defining these events, and then add more and more as we categorize them. I am thinking about Jenkins, Tekton, Travis, Spinnaker, Circle CI, Jenkins X, which doesn't have these advanced concepts, as far as I am aware. 
@e-backmark-ericsson @afrittoli what do you think?

Reply:
I guess from https://github.com/cdfoundation/sig-events/pull/23#discussion_r594263666 we can close this?

Reply:
I'd prefer we use the term "published" instead of "released" here, to not confuse with any discussion on trust or similar. To me "released" is based on a business decision and is somehow stamped with some quality aspect on it, while "published" just means that is has been uploaded to some kind of open repository.

Reply:
@e-backmark-ericsson totally agree with "published"